### PR TITLE
Add VIP_GO_APP_ID to telemetry event data

### PR DESCRIPTION
**Pull request by alecgeatches**
## Description

This PR extends the recent Tracks telemetry work (https://github.com/Automattic/vip-go-mu-plugins/pull/5869, https://github.com/Automattic/vip-go-mu-plugins/pull/5921) to include VIP app IDs in a separate field on events. We already utilize app IDs for user identification:

https://github.com/Automattic/vip-go-mu-plugins/blob/7508ea94d7fdaecf603a73ac46fb36fdd8829560/telemetry/tracks/class-tracks-event.php#L206-L214

This tracks `userids` with values in the format `{{site_id}}_{{wp_user_id}}`. This is less useful for tracking per-app events, since we'd require parsing and preprocessing event data. This PR serves to explicitly track app IDs in a separate data point to make filtering easier. Note that we already track [the app environment](https://github.com/Automattic/vip-go-mu-plugins/blob/7508ea94d7fdaecf603a73ac46fb36fdd8829560/telemetry/tracks/class-tracks-event.php#L156-L162) (e.g. `local`/`develop`) as well as [organization ID](https://github.com/Automattic/vip-go-mu-plugins/blob/7508ea94d7fdaecf603a73ac46fb36fdd8829560/telemetry/tracks/class-tracks-event.php#L164-L170), so this fits right in.

### Changed

- Added `VIP_GO_APP_ID` to Tracks telemetry events

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally or in Codespaces (or has an appropriate fallback).
- [ ] This change works and has been tested on a sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test

Run unit tests:

```bash
./bin/test.sh --filter Tracks_Event_Test
```

### CODE REVIEW

**Comment by alecgeatches** [comment]
I branched these changes from https://github.com/Automattic/vip-go-mu-plugins/pull/5921, so probably best to merge those first.

**Comment by mjangda** [comment]
> We already utilize app IDs for user identification

Should we just remove that block? On VIP, the user id should be set based on the hashed email (the VIP_TELEMETRY_SALT block above).

File: telemetry/tracks/class-tracks-event.php [hunk 1]
```diff
@@ -161,10 +161,18 @@ protected function process_properties(
...
+				$event->vipgo_app = $app_id;
+			}
+		}
+
 		// Set VIP organization if it exists.
 		if ( defined( 'VIP_ORG_ID' ) ) {
 			$org_id = constant( 'VIP_ORG_ID' );
-			if ( is_string( $org_id ) && '' !== $org_id ) {
+			if ( is_integer( $org_id ) && $org_id > 0 ) {
 				$event->vipgo_org = $org_id;
```
Comment (mjangda) [inline telemetry/tracks/class-tracks-event.php]: Similar to the app change, let's rename this:

```suggestion
				$event->vip_org_id = $org_id;
```

File: telemetry/tracks/class-tracks-event.php [hunk 2]
```diff
@@ -161,10 +161,18 @@ protected function process_properties(
 			}
 		}
 
+		// Set VIP app ID if it exists.
+		if ( defined( 'VIP_GO_APP_ID' ) ) {
+			$app_id = constant( 'VIP_GO_APP_ID' );
+			if ( is_integer( $app_id ) && $app_id > 0 ) {
+				$event->vipgo_app = $app_id;
```
Comment (mjangda) [inline telemetry/tracks/class-tracks-event.php]: The constant is incorrectly named :(

Let's rename the prop to `vip_env_id`: 

- remove `go`
- replace `app` with `env`
- add `_id`

```suggestion
			// APP_ID is incorrectly named; it actually references the environment ID
			$env_id = constant( 'VIP_GO_APP_ID' );
			if ( is_integer( $env_id ) && $env_id > 0 ) {
				$event->vip_env_id = $env_id;
```

**Review by mjangda — CHANGES_REQUESTED** [review]
Left some suggestions on this.

It would be nice to cleanup the other props as well (i.e. change any references of `vipgo_` to `vip_` (e.g. `vipgo_env`), although I realize that is increasing the scope of this PR :)

Separately, what do you think about adding the home or site URL as a prop? (`vip_home_url`). The ID is helpful, but the URL would make sorting through data easier.

File: tests/telemetry/tracks/test-class-tracks-event.php [hunk 3]
```diff
@@ -60,6 +63,7 @@ public function test_should_return_event_data() {
 		$this->assertSame( hash_hmac( 'sha256', $this->user->user_email, self::VIP_TELEMETRY_SALT ), $event->get_data()->_ui );
 		$this->assertSame( 'vip:user_email', $event->get_data()->_ut );
 		$this->assertSame( self::VIP_GO_APP_ENVIRONMENT, $event->get_data()->vipgo_env );
+		$this->assertSame( self::VIP_GO_APP_ID, $event->get_data()->vipgo_app );
```
Comment (sanmai) [inline tests/telemetry/tracks/test-class-tracks-event.php]:
```suggestion
		$this->assertSame( self::VIP_GO_APP_ID, $event->get_data()->vip_env_id );
```

File: telemetry/tracks/class-tracks-event-dto.php [hunk 4]
```diff
@@ -41,5 +41,7 @@ class Tracks_Event_DTO {
 
 	public int $vipgo_org;
 
+	public int $vipgo_app;
```
Comment (sanmai) [inline telemetry/tracks/class-tracks-event-dto.php]:
```suggestion
	public int $vip_env_id;
```

File: telemetry/tracks/class-tracks-event.php [hunk 5]
```diff
@@ -161,6 +161,15 @@ protected function process_properties(
 			}
 		}
 
+		// Set VIP app ID if it exists.
```
Comment (sanmai) [inline telemetry/tracks/class-tracks-event.php]:
```suggestion
		// Set VIP environment ID if it exists.
```

**Review by sanmai — APPROVED** [review]
It sounds like we addressed all @mjangda's suggestions.

**Comment by sanmai** [comment]
WP nightly tests are broken, probably unrelated to this PR.

**Comment by alecgeatches** [comment]
It appears these changes were wrapped into the work in #5990 and #5966, [see discussion](https://github.com/Automattic/vip-go-mu-plugins/pull/5966/files#r1841536378). Closing!

**Pull request closed without merging**